phylanx/ir/dictionary.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2018 Weile Wei
+// Copyright (c) 2018 Parsa Amini
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(PHYLANX_DICTIONARY)
+#define PHYLANX_DICTIONARY
+
+#include <phylanx/config.hpp>
+#include <phylanx/util/variant.hpp>
+
+#include <hpx/include/serialization.hpp>
```

Comment:
Why is this header #include needed?

Reply:
@hkaiser I noticed that you included three `#include` headers in your comment, do you specifically point to `#include <hpx/include/serialization.hpp>`? I thought I just copied them from somewhere but I like to investigate it further.

Reply:
I think `#include <hpx/include/serialization.hpp> ` is not needed here as your header file does not refer to anything related to serialization. We should try limiting the used `#include`s to what is needed by a particular file.